javaparser-core-testing/src/test/java/com/github/javaparser/ast/validator/Java16ValidatorTest.java

```diff
@@ -57,7 +57,7 @@ void yieldAllowed() {
     class Record {
 
         @Nested
-        class RecordAsIdentifierForbidden {
+        class RecordAsIdentifierSometimesForbidden {
```

Comment:
Please keep the tests within the `Java16ValidatorTest` consistent with the ones within the Java14/15 preview validator tests.

I fear I misled you within the issue -- my mistake. Perhaps rename this to `RecordAsTypeIdentifierForbidden`?

Additionally, you have correctly pointed out to me that records used as e.g. field and method identifiers is fine -- move the `recordUsedAsFieldName()` test method outside of this?